lib/Dialect/TorchConversion/IR/TorchConversionOps.cpp

```diff
@@ -91,7 +91,20 @@ OpFoldResult FromI64Op::fold(FoldAdaptor adaptor) {
 OpFoldResult ToI64Op::fold(FoldAdaptor adaptor) {
   auto attr = adaptor.getOperand().dyn_cast_or_null<mlir::IntegerAttr>();
   if (attr) {
-    return attr;
+    Attribute foldResult = attr;
+    if (attr.getType().isSignedInteger()) {
+      // convert sint type to i64 type
+      foldResult =
+          IntegerAttr::get(IntegerType::get(getContext(), 64), attr.getSInt());
+    } else if (attr.getType().isUnsignedInteger()) {
+      // convert uint to i64 type
+      foldResult = IntegerAttr::get(IntegerType::get(getContext(), 64),
+                                    static_cast<int64_t>(attr.getUInt()));
```

Comment:
This one is fishy - you need something like https://stackoverflow.com/a/17251989 to make sure that it fits - but I don't trust that particular snippet and I don't know have a hardened implementation on-hand. 

If you want to land without this, I think the rest is good.